src/honey_badger.rs

```diff
@@ -295,16 +323,26 @@ where
         }
 
         // Deserialize the output.
+        let mut faults = FaultLog::new();
         let transactions: BTreeMap<NodeUid, Vec<Tx>> = self
             .decrypted_selections
             .iter()
             .flat_map(|(proposer_id, ser_batch)| {
-                // If deserialization fails, the proposer of that batch is faulty. Ignore it.
-                bincode::deserialize::<Vec<Tx>>(&ser_batch)
-                    .ok()
-                    .map(|proposed| (proposer_id.clone(), proposed))
+                // If deserialization fails, the proposer of that batch is
+                // faulty. Log the faulty proposer and ignore the batch.
+                if let Ok(proposed) = bincode::deserialize::<Vec<Tx>>(&ser_batch) {
+                    Some((proposer_id.clone(), proposed))
+                } else {
+                    let fault_kind = FaultKind::BatchDeserializationFailed;
```

Comment:
I put `let fault_kind` on its own line because:

```rust
faults.append(proposer_id, FaultKind::BatchDeserializationFailed);
```

throws a line-length formatting warning, and rustfmt wants to change it to:

```rust
faults
    .append(proposer_id, FaultKind::BatchDeserializationFailed);
```

which looks weird to me, so I changed most of the `fault_kind`s to be created on their own line:

```rust
let fault_kind = FaultKind::BatchDeserializationFailed;
faults.append(node_id, fault_kind);
```

Do you think 

```rust
faults
    .append(proposer_id, FaultKind::BatchDeserializationFailed);
```

looks better? If so, I'll change it.

Reply:
If `rustfmt` does that then let's just keep the `let`.